runtime/ddr/CMakeLists.txt

```diff
@@ -57,6 +57,8 @@ if(OMR_TOOLCONFIG STREQUAL "gnu")
 	omr_remove_flags(CMAKE_CXX_FLAGS ${OMR_STRNCPY_FORTIFY_OPTIONS})
 	if(NOT CMAKE_CXX_COMPILER_ID MATCHES Clang)
 		target_compile_options(j9ddr_misc PRIVATE -femit-class-debug-always)
+	else()
+		target_compile_options(j9ddr_misc PRIVATE -fno-eliminate-unused-debug-types)
 	endif()
```

Comment:
If we need to add that option, the `if` should be inverted to avoid the negation.
Do we ever have both `OMR_TOOLCONFIG STREQUAL "gnu"` and `CMAKE_CXX_COMPILER_ID MATCHES Clang`?

Reply:
I'll invert the if.

Based on what happens in

https://github.com/eclipse/omr/blob/master/cmake/modules/OmrDetectSystemInformation.cmake#L192-L200

we can have both be true at the same time. Actually, looking at that code, I realize we might build with clang on some platforms already - I should clarify that this PR is one of the changes that allows clang to build JDK21 on Linux.

Reply:
Sorry, that's the C compiler ID detection I linked.

Reply:
I imagine the `CMAKE_CXX_COMPILER_ID` and `CMAKE_C_COMPILER_ID` would normally match, though. In any case, compiling with clang on Linux is definitely a situation where both `OMR_TOOLCONFIG STREQUAL "gnu"` and `CMAKE_CXX_COMPILER_ID MATCHES Clang`, because without this commit `j9ddr` compilation will fail with a bunch of fatal `Unhandled structure type:` errors.

Reply:
In hindsight, it was probably a mistake to use `OMR_TOOLCONFIG` to decide whether to add `-femit-class-debug-always`. Instead, I think we should use `CMAKE_C_COMPILER_ID`:
```
if(OMR_TOOLCONFIG STREQUAL "gnu")
	# defining _FORTIFY_SOURCE is invalid when optimizations are disabled
	omr_remove_flags(CMAKE_C_FLAGS ${OMR_STRNCPY_FORTIFY_OPTIONS})
	omr_remove_flags(CMAKE_CXX_FLAGS ${OMR_STRNCPY_FORTIFY_OPTIONS})
endif()

if(CMAKE_C_COMPILER_ID MATCHES "Clang")
	target_compile_options(j9ddr_misc PRIVATE -fno-eliminate-unused-debug-types)
elseif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
	target_compile_options(j9ddr_misc PRIVATE -femit-class-debug-always)
endif()
```

Reply:
Looking at that `_OMR_TOOLCONFIG` detection again, on Windows the `CMAKE_C_COMPILER_ID` might match `"^(Apple)?Clang$"` but the `_OMR_TOOLCONFIG` will be set to `"msvc"`, so https://github.com/eclipse-openj9/openj9/pull/19807#discussion_r1665798982 might make the rule match more platforms than what I have now. I don't know the details of Windows compilation (or macOS) so that could be harmless.

Reply:
Force-pushed to implement your proposed alternative, as it does seem reasonable.